### DISCUSSION TITLE
Cannot commit multiple files in Mac GitHub Desktop

### DISCUSSION THREAD

**Discussion by splorp** [community/community#24081] [Apps, API and Webhooks]
<p>Since updating my the macOS version of GitHub Desktop to 2.6.6, I can no longer commit multiple files. If I select a single file, the “Commit to master” button is active, but if a select one or more additional file to commit, the button becomes disabled. (See attached screenshots.)</p>
<p>I am running macOS High Sierra 10.13.6 and have verified this behaviour on two different Mac laptops.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/7/179b0dee000bbbf220e7785646b455c9a80be125.png" data-download-href="/uploads/short-url/3mPbqHZLbVdASJNMeOTaGH4WCcl.png?dl=1" title="github-desktop-commit" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/29835/181081848-e1ef1774-cb22-4a54-bd05-db30fd847607.png" alt="github-desktop-commit" data-base62-sha1="3mPbqHZLbVdASJNMeOTaGH4WCcl" width="363" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/7/179b0dee000bbbf220e7785646b455c9a80be125_2_363x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/7/179b0dee000bbbf220e7785646b455c9a80be125_2_544x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/1/7/179b0dee000bbbf220e7785646b455c9a80be125_2_726x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/7/179b0dee000bbbf220e7785646b455c9a80be125_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">github-desktop-commit</span><span class="informations">1920×2640 446 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by steveward** [comment, marked as answer]
<p><a class="mention" href="/splorp">@splorp</a> when selecting multiple files you need to enter a commit message in the <code>Summary (required)</code> text box. Git requires a commit message for commits, and with only one file selected GitHub Desktop pre-fills the commit message. I hope that helps!</p>

**Comment by splorp** [comment]
<p>Hoo, boy.</p>
<p>I’ve been using this app since day one and I don’t think I’ve ever noticed that behaviour before. It makes perfect sense, though. I must have been adding a commit message prior to selecting files from the list of changes prior to thinking this was a UI issue.</p>
<p>Thanks for pointing out this obviousness. :rofl:</p>

**Comment by CheapSquier** [comment]
<p>Ok, just a thought- it seems like the behavior of the commit summary in the desktop is a little atypical. I know from using the command line that I need a comment, so when I saw “Summary (required)”, I typed “Initial commit” into the text box (the larger, lower one). But, the commit button was still disabled. I clicked on the Summary (required) text box and tried backspace and delete. Nothing happens, the text doesn’t change.  I figured maybe that was fixed text and started looking for solutions elsewhere. Only later did I actually try typing something in there that I saw you could change the text and then the Commit button was enabled.</p>
<p>I mostly post this as a hint for other users that are stuck like me. Clearly, based on the number of threads I found on the subject, people are having issues with this. Maybe add an “X” button on the right side of the Summary (required) text box? Maybe hover text that says “Text in the upper box required, text in the lower box, optional”. Maybe move the “Summary (required)” text out of the box and make it a label?</p>
<p>Thanks, and I hope this helps somebody.</p>

**Comment by steveward** [comment]
<p>Thanks for the feedback <a class="mention" href="/cheapsquier">@CheapSquier</a>, I’ll pass that along to the team!</p>
